src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -126,20 +127,27 @@ public async Task<(Error, GitHubUser)> GetByCommit(string authorEmail, string re
             return await GetByLogin(login);
         }
 
-        public Task SaveChanges()
+        public async Task SaveChanges(Config config)
         {
             if (!_updated)
             {
-                return Task.CompletedTask;
+                return;
             }
 
+            string file;
             lock (_lock)
             {
-                var file = new GitHubUserCacheFile
+                file = JsonUtility.Serialize(new GitHubUserCacheFile
                 {
                     Users = Users.ToArray(),
-                };
-                return ProcessUtility.WriteFile(_cachePath, JsonUtility.Serialize(file));
+                });
+            }
+            await ProcessUtility.WriteFile(_cachePath, file);
+            if (config.GitHub.UpdateRemoteUserCache && HrefUtility.IsHttpHref(config.GitHub.UserCache))
+            {
+                // TOOD: aware of ETag to handle conflicts
+                // TODO: error handling
```

Comment:
can these TODOs be handled in this PR? otherwise it is very likely to have unhandled exception.

Reply:
OK. added error handling